src/test/java/seedu/recipe/model/recipe/NameContainsKeywordsPredicateTest.java

```diff
@@ -43,44 +51,56 @@ public void equals() {
     @Test
     public void test_nameContainsKeywords_returnsTrue() {
         // One keyword
-        NameContainsKeywordsPredicate predicate =
-            new NameContainsKeywordsPredicate(Collections.singletonList("Lasagna"));
+        PropertyNameContainsKeywordsPredicate<Name> predicate =
+            new PropertyNameContainsKeywordsPredicate<Name>(Collections.singletonList("Lasagna"),
+                FindUtil.GET_NAME_FROM_RECIPE,
+                FindUtil.GET_NAME_STRING);
         assertTrue(predicate.test(new Recipe(new Name("Lasagna"))));
 
         // Multiple keywords
-        predicate = new NameContainsKeywordsPredicate(Arrays.asList("Grilled", "Cheese"));
+        predicate = new PropertyNameContainsKeywordsPredicate<Name>(Arrays.asList("Grilled", "Cheese"),
+            FindUtil.GET_NAME_FROM_RECIPE,
+            FindUtil.GET_NAME_STRING);
         assertTrue(predicate.test(new Recipe(new Name("Grilled Cheese"))));
 
         // Only one matching keyword
-        predicate = new NameContainsKeywordsPredicate(Arrays.asList("Steamed", "Baked"));
+        predicate = new PropertyNameContainsKeywordsPredicate<Name>(Arrays.asList("Steamed", "Baked"),
+            FindUtil.GET_NAME_FROM_RECIPE,
+            FindUtil.GET_NAME_STRING);
         assertTrue(predicate.test(new Recipe(new Name("Baked Rice"))));
 
         // Mixed-case keywords
-        predicate = new NameContainsKeywordsPredicate(Arrays.asList("pork", "hAlaL"));
+        predicate = new PropertyNameContainsKeywordsPredicate<Name>(Arrays.asList("pork", "hAlaL"),
```

Comment:
Did this keyword dirty smh

Reply:
Please leave it out so potential reviewers don't take offence